runtime/src/main/java/org/corfudb/runtime/view/replication/IReplicationProtocol.java

```diff
@@ -92,8 +94,8 @@
     default @Nonnull
     Map<Long, ILogData> readRange(RuntimeLayout runtimeLayout, Set<Long> globalAddresses) {
         return globalAddresses.parallelStream()
-                .map(a -> new AbstractMap.SimpleImmutableEntry<>(a, read(runtimeLayout, a)))
-                .collect(Collectors.toMap(r -> r.getKey(), r -> r.getValue()));
+                .map(a -> new SimpleImmutableEntry<>(a, read(runtimeLayout, a)))
```

Comment:
it seems to me you don't need `map` here

Reply:
done